apps/services/user-profile/src/app/v2/islykill.service.ts

```diff
@@ -15,20 +53,64 @@ export class IslykillService {
     email?: string
     phoneNumber?: string
   }): Promise<PublicUser> {
-    return this.islyklarApi
-      .islyklarPut({
-        user: {
-          ssn: nationalId,
-          ...(email && { email }),
-          ...(phoneNumber && { mobile: phoneNumber }),
-        },
+    try {
+      const user = {
+        ssn: nationalId,
+        ...(email && { email }),
+        ...(phoneNumber && { mobile: phoneNumber }),
+      }
+
+      return await this.islyklarApi.islyklarPut({ user })
+    } catch ({ message }) {
+      throw new InternalServerErrorException(
+        'Unable to update islykill settings for user',
+        message,
+      )
+    }
+  }
+
+  async createIslykillSettings(
+    nationalId: string,
+    { email, mobile }: { email?: string; mobile?: string },
+  ): Promise<PublicUser> {
+    try {
+      const user = {
+        ssn: nationalId,
+        ...(email && { email }),
+        ...(mobile && { mobile }),
+      }
+
+      return await this.islyklarApi.islyklarPost({ user })
+    } catch ({ message }) {
+      throw new InternalServerErrorException(
+        message,
+        'Unable to create islykill settings for user',
+      )
+    }
+  }
+
+  async upsertIslykillSettings({
+    nationalId,
+    email,
+    phoneNumber,
+  }: {
+    nationalId: string
+    email?: string
+    phoneNumber?: string
+  }): Promise<PublicUser> {
+    const { noUserFound } = await this.getIslykillSettings(nationalId)
+
+    if (noUserFound) {
+      return await this.createIslykillSettings(nationalId, {
```

Comment:
nit: Maybe have the same inputs for create and update methods, then you can reuse their input types